src/header/common/link.rs

```diff
@@ -0,0 +1,1027 @@
+    Stylesheet,
+    /// subsection.
+    Subsection,
+    /// successor-version.
+    SuccessorVersion,
+    /// up.
+    Up,
+    /// versionHistory.
+    VersionHistory,
+    /// via.
+    Via,
+    /// working-copy.
+    WorkingCopy,
+    /// working-copy-of.
+    WorkingCopyOf,
+    /// ext-rel-type.
+    ExtRelType(String)
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Struct methods
+////////////////////////////////////////////////////////////////////////////////
+
+impl Link {
+    /// Create `Link` from a `Vec<LinkValue>`.
+    pub fn new(link_values: Vec<LinkValue>) -> Link {
+        Link { values: link_values }
+    }
+}
+
+#[allow(dead_code)]
+impl LinkValue {
+    /// Create `LinkValue` from URI-Reference.
+    pub fn new<T>(uri: T) -> LinkValue
+        where T: Into<Cow<'static, str>> {
+        LinkValue {
+            link: uri.into(),
+            rel: None,
+            anchor: None,
+            rev: None,
+            href_lang: None,
+            media_desc: None,
+            title: None,
+            title_star: None,
+            media_type: None,
+            link_extension: None,
+        }
+    }
+
+    /// Get the `LinkValue`'s value.
+    pub fn link(&self) -> &str {
+        self.link.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `rel` parameter(s).
+    pub fn rel(&self) -> Option<&Vec<RelationType>> {
+        self.rel.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `anchor` parameter.
+    pub fn anchor(&self) -> Option<&String> {
+        self.anchor.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `rev` parameter(s).
+    pub fn rev(&self) -> Option<&Vec<RelationType>> {
+        self.rev.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `hreflang` parameter(s).
+    pub fn href_lang(&self) -> Option<&Vec<LanguageTag>> {
+        self.href_lang.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `media` parameter(s).
+    pub fn media_desc(&self) -> Option<&Vec<MediaDesc>> {
+        self.media_desc.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `title` parameter.
+    pub fn title(&self) -> Option<&String> {
+        self.title.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `title*` parameter.
+    pub fn title_star(&self) -> Option<&String> {
+        self.title_star.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `type` parameter.
+    pub fn media_type(&self) -> Option<&Mime> {
+        self.media_type.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `link-extension` parameter.
+    pub fn link_extension(&self) -> Option<&String> {
+        self.link_extension.as_ref()
+    }
+
+    /// Add a `RelationType` to the `LinkValue`'s `rel` parameter.
+    pub fn push_rel(mut self, rel: RelationType) -> LinkValue {
+        let mut v = self.rel.take().unwrap_or(Vec::new());
+
+        v.push(rel);
+
+        self.rel = Some(v);
+
+        self
+    }
+
+    /// Set `LinkValue`'s `anchor` parameter.
+    pub fn set_anchor(mut self, anchor: &str) -> LinkValue {
```

Comment:
Perhaps `T: Into<String>` here, so people who allocated a `String` anyways can just give it to us, instead of us having to copy it again.